framework/src/org/checkerframework/framework/test/ImmutableTestConfiguration.java

```diff
@@ -0,0 +1,87 @@
+package org.checkerframework.framework.test;
+
+import org.checkerframework.framework.util.PluginUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Represents all of the information needed to execute the Javac compiler for a given set of test files.
+ */
+public class ImmutableTestConfiguration implements TestConfiguration {
+
+    /**
+     * Options that should be passed to the compiler
```

Comment:
Some options have arguments and some do not, e.g.,
-AprintAllQualifiers
vs.
-Aclasspath "dir1:dir2"
While you can argue that second case is more like 2 arguments, the point of the map is so I can do things like:

addToClasspath("dir1")
addToClasspath("dir2")

rather than
setClasspath("dir1")
setClasspath(getClasspath() + File.pathSeparator" + "dir2")

It is a pretty naive approach to adding options but works pretty well in most cases.  Since we are using it in a limited set of cases in the test suite I am not sure we need to make it any more sophisticated.